api/osf_api.py

```diff
@@ -530,15 +539,28 @@ def create_preprint(
     # (ex: 'osfstorage/627a5b3ab4f587000aa2725c').  So we need to parse out the
     # actual file id to use in the Preprint patch.
     file_id = metadata['data']['id'].split('/')[1]
-    # Step 3: Attach the file to the Preprint and set it as Published
+    # Get the moderation type for the Preprint Provider. If the preprint provider
+    # uses a moderation workflow (either pre-moderation or post-moderation) then
+    # set the publish indicator to False. A pre-moderation preprint cannot be
+    # published until after it has been accepted by the moderator, and a post-
+    # moderation preprint will automatically get published upon creation of the
+    # submit review_action below.
+    mod_type = get_moderation_type_for_provider(
+        session,
+        provider_type='preprints',
+        provider_id=provider_id,
+    )
+    if mod_type:
+        publish_ind = False
+    else:
+        publish_ind = True
```

Comment:
Golfing: `publish_ind = not mod_type`

Reply:
The actual values of mod_type are: "pre-moderation", "post-moderation", or Null. So the if condition is really checking that if the mod_type is not null then set the publish_ind to False. So would it be better to change the statement to "if mod_type is not None:"?

Reply:
`publish_ind = mod_type is None`

Reply:
Fixed.